src/components/WorkspaceSubmissionStatusIcon.ts

```diff
@@ -0,0 +1,97 @@
+import { TooltipTrigger } from '@terra-ui-packages/components';
+import { cond, switchCase } from '@terra-ui-packages/core-utils';
+import { isAfter, parseJSON } from 'date-fns/fp';
+import _ from 'lodash/fp';
+import { ReactNode } from 'react';
+import { div, h, span } from 'react-hyperscript-helpers';
+import { DelayedRender } from 'src/components/common';
+import { icon, spinner } from 'src/components/icons';
+import colors from 'src/libs/colors';
+import { WorkspaceWrapper as Workspace } from 'src/libs/workspace-utils';
+
+export type WorkspaceSubmissionStatus = 'success' | 'failure' | 'running';
+
+export const workspaceSubmissionStatus = (workspace: Workspace): WorkspaceSubmissionStatus | undefined => {
+  const stats = workspace.workspaceSubmissionStats;
+  if (!stats) return undefined;
+  const { runningSubmissionsCount, lastSuccessDate, lastFailureDate } = stats;
+  return cond(
+    [!!runningSubmissionsCount, () => 'running'],
+    [
+      !!lastSuccessDate && (!lastFailureDate || isAfter(parseJSON(lastFailureDate), parseJSON(lastSuccessDate))),
+      () => 'success',
+    ],
+    [!!lastFailureDate, () => 'failure']
+  );
+};
+
+interface WorkspaceSubmissionStatusIconProps {
+  status: WorkspaceSubmissionStatus | undefined;
```

Comment:
Nit: As-is, this prop is required and may be set to `undefined`. Should it be optional? `propName?: Type` would the prop to be set to `undefined` or omitted entirely.

Reply:
I started with trying to make undefined as a part of the type, since the actual function can return it, but that caused all sorts of other issues: so that's how it got there.
You're right though - it is more idiomatic to make it optional.